### ISSUE TITLE
Problem when combining windows generated data on linux machine

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#618]
*Originally reported by* **Anonymous**

----------------------------------------

If i have the following directory structure:

```
- project/
    - module/
        - tests/
            - file.py
```

a coveragerc with the folowing paths configuration:

```
[paths]
source =
    project/module
    */project/module
    *\project\module
```

The windows data file contains:

```
!coverage.py: This is a private format, don't read it directly!{"lines":{"C:\\a\\path\\somewhere\\coveragepy_test\\project\\module\\tests\\file.py":[1,2,3,4]}}
```

When i run, on a Linux machine:

```
$ coverage combine coverage.win.paths
```

the resulting coverage data file is:

```
!coverage.py: This is a private format, don't read it directly!{"lines":{"/tmp/coveragepy_test/project/module/tests\\file.py":[1,2,3,4]}}
```

I tried many different [paths] configuration with no luck.

I'm uploading a .tar.gz with the same scenario so you only need to do coverage combine coverage.win.paths to see the problem happening.

let me know if I can provide more information.

thanks!

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/618
- This issue had attachments: coveragepy_test_combine.tar.gz.  See the [original issue](https://bitbucket.org/ned/coveragepy/issue/618) for details.

**Comment by nedbat** [comment]
*Original comment by* **Bruno Gola (Bitbucket: [brunogola](https://bitbucket.org/brunogola), GitHub: Unknown)**

----------------------------------------

sorry, I didn't notice I wasn't logged in when submitting the issue.

**Comment by nedbat** [comment]
@brunogola Can you say more about what the actual problem is?  Where do things go wrong?

**Comment by nedbat** [comment]
This sounds like something having to do with `[paths]`.  I can help figure out what's going wrong if you link to the code, etc.

**Comment by nedbat** [comment]
*Original comment by* **Bruno Gola (Bitbucket: [brunogola](https://bitbucket.org/brunogola), GitHub: Unknown)**

----------------------------------------

hi @nedbat, the problem is that the resulting data file has a wrong path ("**/tmp/coveragepy_test/project/module/tests\\file.py**")

if you download the .tar file and run (on linux):

```
#!bash
coverage combine coverage.win.path
```

check the .**coverage** file.

there is a .coveragerc in the .tar, i believe the [paths] setting is right.

thanks for the reply

**Comment by nedbat** [comment]
OK, I finally understand: the specific wrong thing is the backslash in the path, since it should be a Linux path.  Interestingly, if you swap the last two lines in your [paths] configuration, it works, though it might not then on Windows.  I'm looking into it.

**Comment by nedbat** [comment]
This is fixed in [87e7ffae9f6f (bb)](https://bitbucket.org/ned/coveragepy/commits/87e7ffae9f6f)

**Comment by nedbat** [comment]
This was included in coverage.py v4.5, released today.
